core/filestorage/filesync/upload.go

```diff
@@ -142,75 +148,91 @@ func (s *fileSync) retryingHandler(ctx context.Context, it *QueueItem) (persiste
 		return persistentqueue.ActionRetry, nil
 	}
 
-	err = s.runOnUploadedHook(it.ObjectId, spaceId)
+	err = s.runOnUploadedHook(it.ObjectId, it.FullFileId())
+	if isObjectDeletedError(err) {
+		return persistentqueue.ActionDone, s.DeleteFile(it.ObjectId, it.FullFileId())
+	}
 	if err != nil {
 		return persistentqueue.ActionRetry, err
 	}
 	s.updateSpaceUsageInformation(spaceId)
 
-	return persistentqueue.ActionDone, s.removeFromUploadingQueues(it)
+	return persistentqueue.ActionDone, s.removeFromUploadingQueues(it.ObjectId)
 }
 
-func (s *fileSync) removeFromUploadingQueues(item *QueueItem) error {
-	err := s.uploadingQueue.Remove(item.Key())
+func (s *fileSync) removeFromUploadingQueues(objectId string) error {
+	if objectId == "" {
+		return nil
+	}
+	err := s.uploadingQueue.Remove(objectId)
 	if err != nil {
 		return fmt.Errorf("remove upload task: %w", err)
 	}
-	err = s.retryUploadingQueue.Remove(item.Key())
+	err = s.retryUploadingQueue.Remove(objectId)
 	if err != nil {
 		return fmt.Errorf("remove upload task from retrying queue: %w", err)
 	}
 	return nil
 }
 
 // UploadSynchronously is used only for invites
-func (s *fileSync) UploadSynchronously(spaceId string, fileId domain.FileId) error {
+func (s *fileSync) UploadSynchronously(ctx context.Context, spaceId string, fileId domain.FileId) error {
 	// TODO After we migrate to storing invites as file objects in tech space, we should update their sync status
 	//  via OnUploadStarted and OnUploaded callbacks
-	err := s.uploadFile(context.Background(), spaceId, fileId)
+	err := s.uploadFile(ctx, spaceId, fileId)
 	if err != nil {
 		return err
 	}
 	s.updateSpaceUsageInformation(spaceId)
 	return nil
 }
 
-func (s *fileSync) runOnUploadedHook(fileObjectId string, spaceId string) error {
-	if s.onUploaded != nil {
-		err := s.onUploaded(fileObjectId)
-		if err != nil && !errors.Is(err, spacestorage.ErrTreeStorageAlreadyDeleted) {
-			log.Warn("on upload callback failed",
-				zap.String("fileObjectId", fileObjectId),
-				zap.String("spaceID", spaceId),
-				zap.Error(err))
-			return err
+func (s *fileSync) runOnUploadedHook(fileObjectId string, fileId domain.FullFileId) error {
+	var merr multierror.Error
```

Comment:
Let's use std errors.Join instead of multierr

Reply:
done